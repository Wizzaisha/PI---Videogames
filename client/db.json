{
  "videogames": [
    {
      "id": 3498,
      "name": "Grand Theft Auto V",
      "released": "2013-09-17",
      "rating": 4.48,
      "background_image": "https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg",
      "playtime": 72,
      "platforms": [
        {
          "id": 4,
          "name": "PC",
          "slug": "pc",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 467216,
          "image_background": "https://media.rawg.io/media/games/618/618c2031a07bbff6b4f611f10b6bcdbc.jpg"
        },
        {
          "id": 186,
          "name": "Xbox Series S/X",
          "slug": "xbox-series-x",
          "image": null,
          "year_end": null,
          "year_start": 2020,
          "games_count": 566,
          "image_background": "https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg"
        },
        {
          "id": 18,
          "name": "PlayStation 4",
          "slug": "playstation4",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 6433,
          "image_background": "https://media.rawg.io/media/games/021/021c4e21a1824d2526f925eff6324653.jpg"
        },
        {
          "id": 16,
          "name": "PlayStation 3",
          "slug": "playstation3",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 3591,
          "image_background": "https://media.rawg.io/media/games/7fa/7fa0b586293c5861ee32490e953a4996.jpg"
        },
        {
          "id": 14,
          "name": "Xbox 360",
          "slug": "xbox360",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 2776,
          "image_background": "https://media.rawg.io/media/games/588/588c6bdff3d4baf66ec36b1c05b793bf.jpg"
        },
        {
          "id": 1,
          "name": "Xbox One",
          "slug": "xbox-one",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 5345,
          "image_background": "https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
        },
        {
          "id": 187,
          "name": "PlayStation 5",
          "slug": "playstation5",
          "image": null,
          "year_end": null,
          "year_start": 2020,
          "games_count": 630,
          "image_background": "https://media.rawg.io/media/games/1f4/1f47a270b8f241e4676b14d39ec620f7.jpg"
        }
      ],
      "genres": [
        {
          "id": 4,
          "name": "Action",
          "slug": "action",
          "games_count": 157488,
          "image_background": "https://media.rawg.io/media/games/511/5118aff5091cb3efec399c808f8c598f.jpg"
        },
        {
          "id": 3,
          "name": "Adventure",
          "slug": "adventure",
          "games_count": 119284,
          "image_background": "https://media.rawg.io/media/games/951/951572a3dd1e42544bd39a5d5b42d234.jpg"
        }
      ]
    },
    {
      "id": 3328,
      "name": "The Witcher 3: Wild Hunt",
      "released": "2015-05-18",
      "rating": 4.67,
      "background_image": "https://media.rawg.io/media/games/618/618c2031a07bbff6b4f611f10b6bcdbc.jpg",
      "playtime": 46,
      "platforms": [
        {
          "id": 7,
          "name": "Nintendo Switch",
          "slug": "nintendo-switch",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 4900,
          "image_background": "https://media.rawg.io/media/games/157/15742f2f67eacff546738e1ab5c19d20.jpg"
        },
        {
          "id": 187,
          "name": "PlayStation 5",
          "slug": "playstation5",
          "image": null,
          "year_end": null,
          "year_start": 2020,
          "games_count": 630,
          "image_background": "https://media.rawg.io/media/games/1f4/1f47a270b8f241e4676b14d39ec620f7.jpg"
        },
        {
          "id": 186,
          "name": "Xbox Series S/X",
          "slug": "xbox-series-x",
          "image": null,
          "year_end": null,
          "year_start": 2020,
          "games_count": 566,
          "image_background": "https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg"
        },
        {
          "id": 1,
          "name": "Xbox One",
          "slug": "xbox-one",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 5345,
          "image_background": "https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
        },
        {
          "id": 4,
          "name": "PC",
          "slug": "pc",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 467216,
          "image_background": "https://media.rawg.io/media/games/618/618c2031a07bbff6b4f611f10b6bcdbc.jpg"
        },
        {
          "id": 18,
          "name": "PlayStation 4",
          "slug": "playstation4",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 6433,
          "image_background": "https://media.rawg.io/media/games/021/021c4e21a1824d2526f925eff6324653.jpg"
        }
      ],
      "genres": [
        {
          "id": 4,
          "name": "Action",
          "slug": "action",
          "games_count": 157488,
          "image_background": "https://media.rawg.io/media/games/511/5118aff5091cb3efec399c808f8c598f.jpg"
        },
        {
          "id": 3,
          "name": "Adventure",
          "slug": "adventure",
          "games_count": 119284,
          "image_background": "https://media.rawg.io/media/games/951/951572a3dd1e42544bd39a5d5b42d234.jpg"
        },
        {
          "id": 5,
          "name": "RPG",
          "slug": "role-playing-games-rpg",
          "games_count": 47991,
          "image_background": "https://media.rawg.io/media/games/d0f/d0f91fe1d92332147e5db74e207cfc7a.jpg"
        }
      ]
    },
    {
      "id": 4200,
      "name": "Portal 2",
      "released": "2011-04-18",
      "rating": 4.61,
      "background_image": "https://media.rawg.io/media/games/328/3283617cb7d75d67257fc58339188742.jpg",
      "playtime": 11,
      "platforms": [
        {
          "id": 14,
          "name": "Xbox 360",
          "slug": "xbox360",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 2776,
          "image_background": "https://media.rawg.io/media/games/588/588c6bdff3d4baf66ec36b1c05b793bf.jpg"
        },
        {
          "id": 6,
          "name": "Linux",
          "slug": "linux",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 69847,
          "image_background": "https://media.rawg.io/media/games/ec3/ec3a7db7b8ab5a71aad622fe7c62632f.jpg"
        },
        {
          "id": 5,
          "name": "macOS",
          "slug": "macos",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 95748,
          "image_background": "https://media.rawg.io/media/games/f46/f466571d536f2e3ea9e815ad17177501.jpg"
        },
        {
          "id": 16,
          "name": "PlayStation 3",
          "slug": "playstation3",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 3591,
          "image_background": "https://media.rawg.io/media/games/7fa/7fa0b586293c5861ee32490e953a4996.jpg"
        },
        {
          "id": 4,
          "name": "PC",
          "slug": "pc",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 467216,
          "image_background": "https://media.rawg.io/media/games/618/618c2031a07bbff6b4f611f10b6bcdbc.jpg"
        },
        {
          "id": 1,
          "name": "Xbox One",
          "slug": "xbox-one",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 5345,
          "image_background": "https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
        }
      ],
      "genres": [
        {
          "id": 2,
          "name": "Shooter",
          "slug": "shooter",
          "games_count": 54820,
          "image_background": "https://media.rawg.io/media/games/16b/16b1b7b36e2042d1128d5a3e852b3b2f.jpg"
        },
        {
          "id": 7,
          "name": "Puzzle",
          "slug": "puzzle",
          "games_count": 89056,
          "image_background": "https://media.rawg.io/media/games/942/9424d6bb763dc38d9378b488603c87fa.jpg"
        }
      ]
    },
    {
      "id": 5286,
      "name": "Tomb Raider (2013)",
      "released": "2013-03-05",
      "rating": 4.05,
      "background_image": "https://media.rawg.io/media/games/021/021c4e21a1824d2526f925eff6324653.jpg",
      "playtime": 11,
      "platforms": [
        {
          "id": 18,
          "name": "PlayStation 4",
          "slug": "playstation4",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 6433,
          "image_background": "https://media.rawg.io/media/games/021/021c4e21a1824d2526f925eff6324653.jpg"
        },
        {
          "id": 5,
          "name": "macOS",
          "slug": "macos",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 95748,
          "image_background": "https://media.rawg.io/media/games/f46/f466571d536f2e3ea9e815ad17177501.jpg"
        },
        {
          "id": 4,
          "name": "PC",
          "slug": "pc",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 467216,
          "image_background": "https://media.rawg.io/media/games/618/618c2031a07bbff6b4f611f10b6bcdbc.jpg"
        },
        {
          "id": 1,
          "name": "Xbox One",
          "slug": "xbox-one",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 5345,
          "image_background": "https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
        },
        {
          "id": 14,
          "name": "Xbox 360",
          "slug": "xbox360",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 2776,
          "image_background": "https://media.rawg.io/media/games/588/588c6bdff3d4baf66ec36b1c05b793bf.jpg"
        },
        {
          "id": 16,
          "name": "PlayStation 3",
          "slug": "playstation3",
          "image": null,
          "year_end": null,
          "year_start": null,
          "games_count": 3591,
          "image_background": "https://media.rawg.io/media/games/7fa/7fa0b586293c5861ee32490e953a4996.jpg"
        }
      ],
      "genres": [
        {
          "id": 4,
          "name": "Action",
          "slug": "action",
          "games_count": 157488,
          "image_background": "https://media.rawg.io/media/games/511/5118aff5091cb3efec399c808f8c598f.jpg"
        },
        {
          "id": 3,
          "name": "Adventure",
          "slug": "adventure",
          "games_count": 119284,
          "image_background": "https://media.rawg.io/media/games/951/951572a3dd1e42544bd39a5d5b42d234.jpg"
        }
      ]
    }
  ]
}
