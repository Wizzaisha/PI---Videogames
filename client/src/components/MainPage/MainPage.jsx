
import { Link, Outlet } from "react-router-dom";
import s from "./MainPage.module.css";

function MainPage () {


    return (
        <div className={s.mainPageContainer}>

            <div className={s.headerContainer}>
                <h2 className={s.navHeadTitle}>Videogames API</h2>
                <p className={s.navHeadText}>Find in our catalog your favorities videogames for your next adventure!</p>
                <div className={s.navBar}>
                    <Link to={"/videogames"}><button className={s.navHeadLink}>Home</button></Link>
                    <Link to={"/videogames/create"}><button className={s.navHeadLink}>Create Videogame</button></Link>
                </div>
            </div>
            <Outlet />
        </div> 
    )
}


export default MainPage;