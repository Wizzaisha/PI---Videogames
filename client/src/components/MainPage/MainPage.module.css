.mainPageContainer {
  margin: 0;
  font-weight: 300;
  gap: 1rem;
}

.headerContainer {
  padding: 3%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  align-items: center;
}

.navHeadText {
  text-align: center;
}

.navHeadTitle {
  font-size: 3rem;
  font-weight: 700;
}

.navBar {
  display: flex;
  justify-content: end;
  gap: 0.2rem;
  margin-right: 2rem;
}

.navBar button {
  height: 3rem;
  width: 9rem;
  background-color: #0c8931cd;
  color: #ededed;
  border-radius: 10px;
}

@media screen and (max-width: 700px) {
  .headerContainer {
    grid-template-columns: 1fr;
    justify-content: center;
    align-items: center;
  }
}
