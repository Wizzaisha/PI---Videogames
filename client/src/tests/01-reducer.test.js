import * as React from "react";
import { create } from "react-test-renderer";

import {
   MemoryRouter,
   Routes,
   Route,
 } from "react-router-dom";
import MainPage from "../components/MainPage/MainPage.jsx";
import LandingPage from "../components/Landing page/LandingPage.jsx";

import rootReducer from "../redux/reducer";
import { 
   GET_ALL_VIDEOGAMES, 
   GET_VIDEOGAME_DETAIL } from "../redux/actions";

import * as data from "../../db.json";

describe("My app", () => {
   it("Renderiza LandingPage correctamente", () => {
     let renderer = create(
       <MemoryRouter initialEntries={["/"]}>
         <Routes>
            <Route path='/' element={<LandingPage />}></Route>
         </Routes>
       </MemoryRouter>
     );
 
     expect(renderer.toJSON()).toMatchSnapshot();
   });

   it("Renderiza la pagina principal correctamente", () => {
      let renderer = create(
         <MemoryRouter initialEntries={["/videogames"]}>
           <Routes>
              <Route path='/videogames' element={<MainPage />}></Route>
           </Routes>
         </MemoryRouter>
       );
   
       expect(renderer.toJSON()).toMatchSnapshot();
   });

 });

 describe("Reducer", () => {
   const state = {
      videogames: [],
      videogamesCopy: [],
      videogameDetails: {},
      createVideogameResponse: {},
      allGenres: [],
      allPlatforms: []
   }

   it("Deberia retornar el estado inicial si no se pasa un type valido", () => {
      expect(rootReducer(undefined, [])).toEqual({
         videogames: [],
         videogamesCopy: [],
         videogameDetails: {},
         createVideogameResponse: {},
         allGenres: [],
         allPlatforms: []
      });
   })

   it("Deberia guardar en nuestro state los videojuegos obtenidos en el back", () => {
      const result = rootReducer(state, {
         type: GET_ALL_VIDEOGAMES,
         payload: data.videogames
      });

      expect(result).not.toEqual(state);
      expect(result).toEqual({
         videogames: data.videogames,
         videogamesCopy: data.videogames,
         videogameDetails: {},
         createVideogameResponse: {},
         allGenres: [],
         allPlatforms: []
      })
   });

   it('DeberÃ­a guardar en nuestro state el product obtenido de nuestro llamado al back cuando action type es "GET_VIDEOGAME_DETAIL"', () => {
      const result = rootReducer(state, {
         type: GET_VIDEOGAME_DETAIL,
         payload: data.videogames[0],
      });
      // Acuerdense que el state inicial no tiene que mutar!
      expect(result).not.toEqual(state);
      expect(result).toEqual({
         videogames: [],
         videogamesCopy: [],
         videogameDetails: data.videogames[0],
         createVideogameResponse: {},
         allGenres: [],
         allPlatforms: []
      });
   });
 })