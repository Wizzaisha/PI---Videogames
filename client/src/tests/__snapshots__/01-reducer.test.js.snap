// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My app Renderiza LandingPage correctamente 1`] = `
<div
  className="landingPageCover"
>
  <div
    className="items"
  >
    <h1
      className="landTitle"
    >
      Videogames page
    </h1>
    <p
      className="landText"
    >
      Find your next adventure here!
    </p>
    <a
      href="/videogames"
      onClick={[Function]}
    >
      <button
        className="landButton"
      >
        To main page
      </button>
    </a>
  </div>
</div>
`;

exports[`My app Renderiza la pagina principal correctamente 1`] = `
<div
  className="mainPageContainer"
>
  <div
    className="headerContainer"
  >
    <h2
      className="navHeadTitle"
    >
      Videogames API
    </h2>
    <p
      className="navHeadText"
    >
      Find in our catalog your favorities videogames for your next adventure!
    </p>
    <div
      className="navBar"
    >
      <a
        href="/videogames"
        onClick={[Function]}
      >
        <button
          className="navHeadLink"
        >
          Home
        </button>
      </a>
      <a
        href="/videogames/create"
        onClick={[Function]}
      >
        <button
          className="navHeadLink"
        >
          Create Videogame
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`My app renders correctly 1`] = `
<div
  className="landingPageCover"
>
  <div
    className="items"
  >
    <h1
      className="landTitle"
    >
      Videogames page
    </h1>
    <p
      className="landText"
    >
      Find your next adventure here!
    </p>
    <a
      href="/videogames"
      onClick={[Function]}
    >
      <button
        className="landButton"
      >
        To main page
      </button>
    </a>
  </div>
</div>
`;
